### DISCUSSION TITLE
Integer-to-character conversion

### DISCUSSION THREAD

**Discussion by HugoMVale** [fortran-lang/stdlib#676] [General]
Hello,
Is there a stdlib function which allows to convert an integer to a character vector (opposite of `ichar` if I understand well)?
I mean, something more or less like:
```fortran
    function itoa(i) result(res)
    !------------------------------------------------------------------------------------------
    !! Convert integer to string.
    !------------------------------------------------------------------------------------------
    integer, intent(in) :: i
        !! Integer
    character(:), allocatable :: res
    character(range(i)+2) :: tmp
      write(tmp,'(i0)') i
      res = trim(tmp)
    end function
```
Thanks for any hint.

**Comment by jvdp1** [comment]
to_string (
https://stdlib.fortran-lang.org/page/specs/stdlib_strings.html#to_string)
is most likely what you are looking for

Le sam. 10 sept. 2022 à 19:18, HugoMVale ***@***.***> a
écrit :

**Reply by HugoMVale** [reply]
Thanks a lot, @jvdp1. Now that you told me where to search for, I realized that the section [Contributing and specs](https://stdlib.fortran-lang.org/page/index.html) has all the necessary explanations, but I never thought of looking under than title... I searched directly under Modules and Procedures and got lost. Perhaps we could rename *Contributing and specs* to *API reference* (as numpy) or similar.
